guidance/models/_azure_openai.py

```diff
@@ -1,12 +1,18 @@
+import pathlib
+import re
+
 from typing import Type
 from urllib.parse import parse_qs, urlparse
 
+import tiktoken
```

Comment:
If we're getting rid of a hard `openai` dependency, we may have to do an optional import check here too

Reply:
`tiktoken` is still a hard dependency.... should that change?